### PR TITLE
Add/update all SDKs with the new sku name ST1/ST2

### PR DESCRIPTION
**Pull request by poanchen**
* Add Java and Go example.
* Change the default sku to be ST2. (applies to all languages)
* Change the default location to be unitedstates. (applies to all languages)
* Made sure all the languages that we support has the following examples, check if resource name is available, create or update an app, retrieve an app, update an app, list all the apps that belong to a specific resource group, and delete an app.
* Add missing README.MD.
* Update armTemplate to use default template.

### CODE REVIEW

File: csharp/README.MD [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# iotcentral-arm-sdk-examples-csharp
+
+Sample code for using [Microsoft.Azure.Management.IotCentral](https://www.nuget.org/packages/Microsoft.Azure.Management.IotCentral/2.1.0) SDK provided by [Microsoft Azure](https://github.com/Azure). This sample code simply check if the resource name is available, create or update an app, retrieve app information, update app, list all the apps that belongs to the specific resource group, and finally delete the app if uncommented it.
```
Comment (emgarten) [inline csharp/README.MD]: minor grammar fixes: This sample code will check if the resource name is available, create or update an app, retrieve app information, update app, list all the apps that belongs to the specific resource group, and finally delete the app if uncommented.

Comment (emgarten) [reply csharp/README.MD]: I see this came from ARM originally, but now that is used for all of them it might be worth updating

Comment (poanchen) [reply csharp/README.MD]: Will update all of them

Comment (poanchen) [reply csharp/README.MD]: Good catch!

**Pull request merged**